src/TLuaInterpreter.cpp

```diff
@@ -10303,86 +10349,131 @@ int TLuaInterpreter::clearSpecialExits(lua_State* L)
 }
 
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getSpecialExits
+// This function was slightly borked - the version in place from 2011 to 2020
+// did not handle the corner case of multiple special exits that go to the same
+// room, it would only show one of them at random. Each special exit was listed
+// in its own table (against the key of the exit room id) and it is a key to a
+// "1" or "0" depending on whether the exit is locked or not. This was not
+// documented in the wiki!
 int TLuaInterpreter::getSpecialExits(lua_State* L)
 {
+    Host& host = getHostFromLua(L);
     if (!lua_isnumber(L, 1)) {
-        lua_pushstring(L, "getSpecialExits: wrong argument type");
+        lua_pushfstring(L, "getSpecialExits: bad argument #1 type (exit room id as number expected, got %s!)", luaL_typename(L, 1));
         return lua_error(L);
     }
     int id_from = lua_tointeger(L, 1);
-
-    Host& host = getHostFromLua(L);
     TRoom* pR = host.mpMap->mpRoomDB->getRoom(id_from);
-    if (pR) {
-        QMapIterator<int, QString> it(pR->getOtherMap());
+    if (!pR) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "room with id %d does not exist", id_from);
+        return 2;
+    }
+
+    bool showAllExits = false;
+    if (lua_gettop(L) > 1) {
+        if (!lua_isboolean(L, 2)) {
```

Comment:
I think it is better to be consistent in how we deal with bad functions: create a `getSpecialExits1` rather than complicating the code, and usage, of the original.

Reply:
It turns out that due to insufficient testing data I didn't find out that the output being displayed (from using `lua display(getSpecialExits())`  was not as wrong as I thought - because I had chosen `1`, `2`, `3` as three different special exit room ids the *list* format output from `display(...)` which did NOT show the keys was ***correct***. The only defect was that for multiple special exits with the same destination it just listed ONE of them **at random**. So the fix - to show (one of) the best (unlocked and lowest exit weighting) one is ***entirely backwards compatible*** when the existing function is used as is...

All that needs to be done is to fix the detail of the example code and to add a warning that:
> "In the corner case where there is **more than one** special exit from this room to the **same** destination room, only one room is shown chosen at random prior to Mudlet **4.y.z** - after that point only the **best**, in terms of being unlocked and of lowest exit weighting (or one of the best if there is a tie) special exits to that room is listed unless the optional `true` argument is given which will cause all of them to be listed."

Reply:
OK, that sounds simple.